### DISCUSSION TITLE
confusion

### DISCUSSION THREAD

**Discussion by Jay-deng** [prometheus/prometheus#11588] [Q&A]
Dear developers, we find that in your program (prometheus/prometheus/util/gate/gate.go:34), the context that ctx.Done() is waiting for has not been canceled, so the program may be blocked all the time. So it will cause the goroutine where the ctx.Done() is located to leak (a partial deadlock). Can you confirm this bug? Thank you.

**Comment by roidelapluie** [comment]
It is not a bug, the gate is meant to be stopped when the context is cancelled. But it should not be cancelled within the gate, as it does not belong to the gate. In prometheus, the context would be cancelled e.g. when Prometheus stops.
